core/src/test/java/apoc/create/CreateTest.java

```diff
@@ -315,6 +318,51 @@ public void testClonePathShouldNotEditOriginalOne() {
                     assertFalse(end.hasProperty("brazorf"));
                 });
     }
+
+    @Test
+    public void testClonePathShouldNotDuplicateRelsWithMultipaths() {
+        //create path with single rels
+        db.executeTransactionally("""
+                          CREATE (n:Node{id:0}),
+                            (n2:Node{id:1}),
+                            (n3:Node{id:2}),
+                            (n4:Node{id:3}),
+                            (n5:Node{id:4}),
+                            (n6:Node{id:5}),
+                            (n7:Node{id:6}),
+                            (n)-[:PARENT]->(n2),
+                            (n)-[:PARENT]->(n3),
+                            (n)-[:PARENT]->(n5),
+                            (n)-[:PARENT]->(n6)-[:PARENT]->(n7),
+                            (n)-[:PARENT]->(n4)-[:PARENT]->(n7)-[:PARENT]->(n2)-[:PARENT]->(n3)"""
+        );
+        
+        // returns a list with all rels
+        testCall(db, """
+                MATCH p=(n:Node{id: 0})-[:PARENT*..4]->(v:Node)
+                WITH collect(p) as p
+                CALL apoc.create.clonePathsToVirtual(p)
+                YIELD path
+                WITH collect( relationships(path) ) as pathRels
+                RETURN apoc.coll.flatten(pathRels) as rels""", r -> {
+            final List<Relationship> paths = (List<Relationship>) r.get("rels");
+            assertFalse(paths.isEmpty());
+
+            // group the rels by the pair of start-end node 
+            final Map<Pair<String, String>, List<Relationship>> collect = paths.stream()
+                    .collect(Collectors.groupingBy(i -> Pair.of(
+                            i.getStartNode().getElementId(),
+                            i.getEndNode().getElementId()))
+                    );
```

Comment:
The verification seems overly complicated to me. With the smaller sample set above, it would be enough to verify that:
* The list of relationship has size 3.
* The collection `Map<Relationship, Long> collection = rels.stream().collect(Collectors.groupingBy(/*Pair*/) has size 2.

Reply:
Changed. 
I'm not really sure if that's the change you meant, because if I'm not mistaken `Map<Relationship, Long>` and `.collect(Collectors.groupingBy(/Pair/)` are incorrect. 
Let me know if as changed is what you wanted, in case I can re-change it :)